### PR TITLE
Added 'Authorization' header with bearer token

### PR DESCRIPTION
**Pull request by rhiroshi**
Added Authorization header with bearer token defined in [RFC 6750](https://datatracker.ietf.org/doc/html/rfc6750#section-2.1) 
related issues: #1522 #902 #1487

### CODE REVIEW

File: app/controllers/devise_token_auth/concerns/set_user_by_token.rb [hunk 1]
```diff
@@ -45,10 +49,10 @@ def set_user_by_token(mapping = nil)
     end
 
     # parse header for values necessary for authentication
-    uid              = request.headers[uid_name] || params[uid_name] || parsed_auth_cookie[uid_name]
+    uid              = request.headers[uid_name] || params[uid_name] || parsed_auth_cookie[uid_name] || decoded_authorization_token[uid_name]
     @token           = DeviseTokenAuth::TokenFactory.new unless @token
-    @token.token     ||= request.headers[access_token_name] || params[access_token_name] || parsed_auth_cookie[access_token_name]
-    @token.client ||= request.headers[client_name] || params[client_name] || parsed_auth_cookie[client_name]
+    @token.token     ||= request.headers[access_token_name] || params[access_token_name] || parsed_auth_cookie[access_token_name] || decoded_authorization_token[access_token_name]
```
Comment (MaicolBen) [inline app/controllers/devise_token_auth/concerns/set_user_by_token.rb:54]: This seems a not breaking way to include authorization header with bearer, great 👏

**Review by MaicolBen — APPROVED** [review]

**Comment by theblang** [comment]
@rhiroshi Could we turn this off when `DeviseTokenAuth.cookie_enabled`? This PR increases the cookie size considerably, and unnecessarily since an `Authorization` header doesn't make sense when using the cookie transport method.

**Comment by rhiroshi** [comment]
commented with a different account. I will create a PR this week to address this

**Comment by barillax** [comment]
@rhiroshi any update on making it possible to disable this header when DeviseTokenAuth.cookie_enabled is true? Thanks!

**Comment by rhiroshi** [comment]
@theblang @barillax Created the PR, sorry for the delay

**Pull request merged**